src/platform/intel/cavs/lib/memory.c

```diff
@@ -112,12 +112,12 @@ static SHARED_DATA struct block_map rt_shared_heap_map[] = {
 #endif
 
 /* Heap blocks for buffers */
-static SHARED_DATA struct block_hdr buf_block[HEAP_BUFFER_COUNT];
+static SHARED_DATA struct block_hdr buf_block[HEAP_BUFFER_COUNT_MAX];
```

Comment:
do you have an idea how much RAM this is wasting?

Reply:
yes, I talked this with @lgirdwood , it is 8B/256B about 3%here.

There are some waste, the percentage is up to how much of it is inevitable: e.g.
On TGL like (~3MB HPSRAM), we have about 2/3 memory used for this BUFFER zone so only 1/3 * 3% = 1% is wasted.
On APL like (~0.5MB HPSRAM), the wasting is worst as there is about 1/3 memory used for this BUFFER zone so about 2/3 * 3% = 2% is wasted.

I previously has some calculation to improve it for small SRAM platforms, like 
```
/* the buffer zone will not occupy more than half of the HP SRAM */
#define HEAP_BUFFER_COUNT_MAX  (HP_SRAM_SIZE / (HEAP_BUFFER_BLOCK_SIZE * 2))
```

This could help to bring back about 5KB memory for us, we can do this kind of fine tune incrementally.